spec/services/reblog_service_spec.rb

```diff
@@ -35,13 +35,24 @@
   end
 
   context 'when the reblogged status is discarded in the meantime' do
-    let(:status) { Fabricate(:status, account: alice, visibility: :public) }
+    let(:status) { Fabricate(:status, account: alice, visibility: :public, text: 'race-condition-discard') }
 
+    # Add a callback to discard the status being reblogged after the
+    # validations pass but before the database commit is executed.
     before do
-      # Update the in-database attribute without reflecting the change in
-      # the object. This cannot simulate all race conditions, but it is
-      # pretty close.
-      Status.where(id: status.id).update_all(deleted_at: Time.now.utc) # rubocop:disable Rails/SkipsModelValidations
+      Status.class_eval do
+        after_validation :discard_status
+        def discard_status
+          Status
+            .where(text: 'race-condition-discard')
```

Comment:
Good idea -- I initially did it this way because the added callback method doesn't have access to vars from the spec, but I can change to just use the actual reblog. Will do and push in a bit.